clients/coredata/event.go

```diff
@@ -131,7 +131,20 @@ func (e *EventRestClient) EventsForDeviceAndValueDescriptor(deviceId string, vd
 
 // Add event
 func (e *EventRestClient) Add(event *models.Event, ctx context.Context) (string, error) {
-	return clients.PostRequest(e.url, []byte(event.String()), ctx)
+	content := clients.FromContext(clients.ContentType, ctx)
+
+	var data []byte
+	var err error
+	if content == clients.ContentTypeCBOR {
+		data, err = event.MarshalCBOR()
+	} else {
+		data, err = event.MarshalJSON()
```

Comment:
You don't want to call this directly. It's an implementation method for the Marshaler interface that is called by `json.Marshal`